tests/integration/telemetry/wasmplugin/wasmplugin_test.go

```diff
@@ -0,0 +1,375 @@
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/retry"
+	common "istio.io/istio/tests/integration/telemetry/stats/prometheus"
+)
+
+var (
+	client, server echo.Instances
+	appNsInst      namespace.Instance
+	registry       registryredirector.Instance
+)
+
+const (
+	// Same user name and password as specified at pkg/test/fakes/imageregistry
+	registryUser   = "user"
+	registryPasswd = "passwd"
+
+	imageName = "istio-testing/wasm/header-injector"
+
+	injectedHeader = "x-resp-injection"
+)
+
+type wasmTestConfigs struct {
+	desc            string
+	name            string
+	policy          string
+	tag             string
+	upstreamVersion string
+	expectedVersion string
+}
+
+var generation = 0
+
+func mapTagToVersionOrFail(t framework.TestContext, tag, version string) {
+	t.Helper()
+	if err := registry.SetupTagMap(map[string]string{
+		imageName + ":" + tag: version,
+	}); err != nil {
+		t.Fatalf("failed to setup the tag map: %v", err)
+	}
+}
+
+func applyAndTestWasm(t framework.TestContext, c wasmTestConfigs) {
+	t.Helper()
+	defer func() {
+		generation++
+	}()
+	scopes.Framework.Infof(c.desc)
+	mapTagToVersionOrFail(t, c.tag, c.upstreamVersion)
+	if err := installWasmExtension(t, c.name, c.tag, c.policy, fmt.Sprintf("g-%q", generation)); err != nil {
+		t.Fatalf("failed to install WasmPlugin: %v", err)
+	}
+	retry.UntilSuccessOrFail(t, func() error {
+		result, err := sendTraffic()
+		if err != nil {
+			t.Log("failed to send traffic")
+			return err
+		}
+
+		if result.Responses.Len() == 0 {
+			return errors.New("failed to get response")
+		}
+
+		value := result.Responses[0].ResponseHeaders.Get(injectedHeader)
+		if value != c.expectedVersion {
+			return fmt.Errorf("[generation: %d] unexpected values for the header %q with the policy %q: given %q, got %q, want %q",
+				generation, injectedHeader, c.policy, c.upstreamVersion, value, c.expectedVersion)
+		}
+
+		return nil
+	}, retry.Delay(1*time.Second), retry.Timeout(100*time.Second))
+}
+
+func resetWasm(t framework.TestContext, pluginName string) {
+	t.Helper()
+	scopes.Framework.Infof("Delete WasmPlugin %q", pluginName)
+	if err := uninstallWasmExtension(t, pluginName); err != nil {
+		t.Fatal(err)
+	}
+
+	successCount := 0
+
+	retry.UntilSuccessOrFail(t, func() error {
+		result, err := sendTraffic()
+		if err != nil {
+			t.Log("failed to send traffic")
+			return err
+		}
+
+		if result.Responses.Len() == 0 {
+			return errors.New("failed to get response")
+		}
+
+		value := result.Responses[0].ResponseHeaders.Get(injectedHeader)
+
+		if value != "" {
+			return fmt.Errorf("failed to reset WasmPlugin %q", pluginName)
+		}
+
+		if successCount < 2 {
+			successCount++
+			return fmt.Errorf("retry")
+		}
+		return nil
+	}, retry.Delay(1*time.Second), retry.Timeout(100*time.Second))
+}
+
+func TestWasmPluginPullPolicy(t *testing.T) {
+	framework.NewTest(t).
+		Features("extensibility.wasm.remote-load").
```

Comment:
test names not match function name?

Reply:
I have changed the function name, but I am not sure I got your point. 
Could you check again, and give a comment if I did not understand?

Reply:
if there's no plan to extend this test case, why not just name it same as function name instead of `remote-load`?

Reply:
I see. I added `image-pull-policy`.